synapse/handlers/room.py

```diff
@@ -732,6 +738,15 @@ async def create_room(
         invite_3pid_list = config.get("invite_3pid", [])
         invite_list = config.get("invite", [])
 
+        # validate each entry for correctness
+        for invite_3pid in invite_3pid_list:
+            if not self._has_all_3pid_keys(invite_3pid):
+                raise SynapseError(
```

Comment:
we should probably add a note in the docstring that SynapseError is also raised if the `config` is malformed.